client/src/main/java/org/spine3/base/Stringifiers.java

```diff
@@ -161,4 +167,129 @@ protected CommandId doBackward(String s) {
+         * That constructor should be used when need to use
+         * a custom delimiter of the elements during conversion.
+         *
+         * @param keyClass   the class of the key elements
+         * @param valueClass the class of the value elements
+         * @param delimiter  the delimiter for the passed elements via string
+         */
+        public MapStringifier(Class<K> keyClass, Class<V> valueClass, String delimiter) {
+            super();
+            this.keyClass = keyClass;
+            this.valueClass = valueClass;
+            this.delimiter = ESCAPE_SEQUENCE + delimiter;
+        }
+
+        @Override
+        protected String doForward(Map<K, V> map) {
+            final String result = map.toString();
+            return result;
+        }
+
+        @Override
+        protected Map<K, V> doBackward(String s) {
+            final String[] buckets = s.split(Pattern.quote(delimiter));
+            final Map<K, V> resultMap = newHashMap();
+
+            for (String bucket : buckets) {
+                saveConvertedBucket(resultMap, bucket);
+            }
+            Ints.stringConverter();
+            return resultMap;
+        }
+
+        private Map<K, V> saveConvertedBucket(Map<K, V> resultMap, String element) {
+            final String[] keyValue = element.split(Pattern.quote(KEY_VALUE_DELIMITER));
+            checkKeyValue(keyValue);
+
+            final String key = keyValue[0];
+            final String value = keyValue[1];
+
+            try {
+                final K convertedKey = convert(keyClass, key);
+                final V convertedValue = convert(valueClass, value);
+                resultMap.put(convertedKey, convertedValue);
+                return resultMap;
+            } catch (Throwable ignored) {
+                throw conversionArgumentException(CONVERSION_EXCEPTION);
+            }
+        }
+
+        @SuppressWarnings("unchecked") // It is OK because class is verified.
+        private static <I> I convert(Class<I> elementClass, String elementToConvert) {
+
+            if (isInteger(elementClass)) {
+                return (I) Ints.stringConverter()
+                               .convert(elementToConvert);
+            }
+
+            if (isLong(elementClass)) {
+                return (I) Longs.stringConverter()
+                                .convert(elementToConvert);
+            }
+
+            if (isString(elementClass)) {
+                return (I) elementToConvert;
+            }
+
+            final I convertedValue = parse(elementToConvert, TypeToken.of(elementClass));
+            return convertedValue;
+
+        }
+
+        private static void checkKeyValue(String[] keyValue) {
+            if (keyValue.length != 2) {
+                final String exMessage =
+                        "Illegal key - value format, key value should be " +
+                        "separated with single `:` character";
+                throw conversionArgumentException(exMessage);
```

Comment:
That branch is tested with the `throw_exception_when_occurred_exception_during_conversion` test method.